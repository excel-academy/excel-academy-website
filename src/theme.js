import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
