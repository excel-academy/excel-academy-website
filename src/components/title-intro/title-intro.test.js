import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FaBriefcaseMedical } from 'react-icons/fa';
import theme from '../../theme';
import TitleIntro from './title-intro';

describe('TitleIntro', () => {
  it('default renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <TitleIntro
          title="hello"
          intro="world"
        />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('default renders with an icon', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <TitleIntro
          title="hello"
          intro="world"
          icon={<FaBriefcaseMedical />}
        />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('default renders with different cols', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <TitleIntro
          title="hello"
          intro="world"
          cols={5}
        />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('default renders with different titleColor', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <TitleIntro
          title="hello"
          intro="world"
          titleColor="black"
        />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
