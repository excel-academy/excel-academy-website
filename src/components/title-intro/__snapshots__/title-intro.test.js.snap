// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleIntro default renders correctly 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 3rem;
  color: #1c2070;
}

.c2 {
  font-size: 1.25rem;
  font-weight: 300;
}

.c0 {
  box-sizing: border-box;
  text-align: center;
}

@media screen and (min-width:46em) {
  .c0 {
    width: 83.33333333333334%;
  }
}

<div
  className="c0"
  width={
    Object {
      "tablet": 0.8333333333333334,
    }
  }
>
  <h1
    className="c1"
    color="purples.3"
  >
    hello
  </h1>
  <p
    className="c2"
  >
    world
  </p>
</div>
`;

exports[`TitleIntro default renders with an icon 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 3rem;
  color: #1c2070;
}

.c3 {
  font-size: 1.25rem;
  font-weight: 300;
}

.c0 {
  box-sizing: border-box;
  text-align: center;
}

.c2 {
  margin-right: 0.5em;
  position: relative;
  top: 2px;
}

@media screen and (min-width:46em) {
  .c0 {
    width: 83.33333333333334%;
  }
}

<div
  className="c0"
  width={
    Object {
      "tablet": 0.8333333333333334,
    }
  }
>
  <h1
    className="c1"
    color="purples.3"
  >
    <span
      className="c2"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128h-80V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v48H48c-26.5 0-48 21.5-48 48v288c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V176c0-26.5-21.5-48-48-48zM192 96h128v32H192V96zm160 248c0 4.4-3.6 8-8 8h-56v56c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-56h-56c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h56v-56c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v56h56c4.4 0 8 3.6 8 8v48z"
        />
      </svg>
    </span>
    hello
  </h1>
  <p
    className="c3"
  >
    world
  </p>
</div>
`;

exports[`TitleIntro default renders with different cols 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 3rem;
  color: #1c2070;
}

.c2 {
  font-size: 1.25rem;
  font-weight: 300;
}

.c0 {
  box-sizing: border-box;
  text-align: center;
}

@media screen and (min-width:46em) {
  .c0 {
    width: 41.66666666666667%;
  }
}

<div
  className="c0"
  width={
    Object {
      "tablet": 0.4166666666666667,
    }
  }
>
  <h1
    className="c1"
    color="purples.3"
  >
    hello
  </h1>
  <p
    className="c2"
  >
    world
  </p>
</div>
`;

exports[`TitleIntro default renders with different titleColor 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 3rem;
  color: black;
}

.c2 {
  font-size: 1.25rem;
  font-weight: 300;
}

.c0 {
  box-sizing: border-box;
  text-align: center;
}

@media screen and (min-width:46em) {
  .c0 {
    width: 83.33333333333334%;
  }
}

<div
  className="c0"
  width={
    Object {
      "tablet": 0.8333333333333334,
    }
  }
>
  <h1
    className="c1"
    color="black"
  >
    hello
  </h1>
  <p
    className="c2"
  >
    world
  </p>
</div>
`;
